### DISCUSSION TITLE
No EX for json.set?

### DISCUSSION THREAD

**Discussion by NavaceSystem** [NodeRedis/node_redis#2754] [Q&A] [answered]
I'm using the node client and also looked the `json.get` docs here https://redis.io/docs/latest/commands/json.set/ and it doesn't seem to support key expiry at set. How am I supposed to set expiry for json keys then? Am I supposed to manually call expire https://redis.io/docs/latest/commands/expire/ for every key I set?

**Comment by leibale** [comment, marked as answer]
Unfortunately `JSON.SET` does not support setting the key expiry.
You can either "pipeline" them one after another:
```
await Promise.all([
  client.set('json', ...),
  client.expire('json', ...)
]);
```
use MULTI:
```
await client.multi()
  .set('json', ...),
  .expire('json', ...)
  .exec();
```
or, you can use a function/lua script.

**Reply by NavaceSystem** [reply]
Yep, that's what I ended up doing. Thanks for confirming.
